Building from source fails with latest version of Godot

### System information

Windows

### Issue description
Building fails with the following error

```
[ 22%] Compiling modules\gdsdecomp\exporters\sample_exporter.cpp ...
[ 22%] C:\...\godot\modules\gdsdecomp\compat\file_access_encrypted_v3.h(86): error C2555: 'FileAccessEncryptedv3::store_8': overriding virtual function return type differs and is not covariant from 'FileAccess::store_8'
.\core/io/file_access.h(201): note: see declaration of 'FileAccess::store_8'
C:\...\godot\modules\gdsdecomp\compat\file_access_encrypted_v3.h(87): error C2555: 'FileAccessEncryptedv3::store_buffer': overriding virtual function return type differs and is not covariant from 'FileAccess::store_buffer'
.\core/io/file_access.h(218): note: see declaration of 'FileAccess::store_buffer'
modules\gdsdecomp\compat\file_access_encrypted_v3.cpp(100): warning C4458: declaration of 'key' hides class member
C:\...\godot\modules\gdsdecomp\compat\file_access_encrypted_v3.h(50): note: see declaration of 'FileAccessEncryptedv3::key'
[ 22%] Compiling modules\gdsdecomp\exporters\scene_exporter.cpp ...
scons: *** [modules\gdsdecomp\compat\file_access_encrypted_v3.windows.editor.x86_64.obj] Error 2
modules\gdsdecomp\compat/file_access_encrypted_v3.h(86): error C2555: 'FileAccessEncryptedv3::store_8': overriding virtual function return type differs and is not covariant from 'FileAccess::store_8'
.\core/io/file_access.h(201): note: see declaration of 'FileAccess::store_8'
modules\gdsdecomp\compat/file_access_encrypted_v3.h(87): error C2555: 'FileAccessEncryptedv3::store_buffer': overriding virtual function return type differs and is not covariant from 'FileAccess::store_buffer'
.\core/io/file_access.h(218): note: see declaration of 'FileAccess::store_buffer'
.\core/typedefs.h(122): warning C4018: '>': signed/unsigned mismatch
.\core/typedefs.h(122): note: the template instantiation context (the oldest one first) is
modules\gdsdecomp\bytecode\bytecode_base.cpp(588): note: see reference to function template instantiation 'auto MAX<int,V>(const T,const T2)' being compiled
        with
        [
            V=uint32_t,
            T=int,
            T2=uint32_t
        ]
modules\gdsdecomp\bytecode\bytecode_base.cpp(760): warning C4018: '>=': signed/unsigned mismatch
scons: *** [modules\gdsdecomp\bytecode\bytecode_base.windows.editor.x86_64.obj] Error 2
modules\gdsdecomp\exporters\oggstr_exporter.cpp(73): warning C4018: '>': signed/unsigned mismatch
C:\...\godot\modules\gdsdecomp\utility/file_access_gdre.h(117): error C2555: 'FileAccessGDRE::store_8': overriding virtual function return type differs and is not covariant from 'FileAccess::store_8'
.\core/io/file_access.h(201): note: see declaration of 'FileAccess::store_8'
C:\...\godot\modules\gdsdecomp\utility/file_access_gdre.h(118): error C2555: 'FileAccessGDRE::store_buffer': overriding virtual function return type differs and is not covariant from 'FileAccess::store_buffer'
.\core/io/file_access.h(218): note: see declaration of 'FileAccess::store_buffer'
scons: *** [modules\gdsdecomp\register_types.windows.editor.x86_64.obj] Error 2
modules\gdsdecomp\compat/file_access_encrypted_v3.h(86): error C2555: 'FileAccessEncryptedv3::store_8': overriding virtual function return type differs and is not covariant from 'FileAccess::store_8'
.\core/io/file_access.h(201): note: see declaration of 'FileAccess::store_8'
modules\gdsdecomp\compat/file_access_encrypted_v3.h(87): error C2555: 'FileAccessEncryptedv3::store_buffer': overriding virtual function return type differs and is not covariant from 'FileAccess::store_buffer'
.\core/io/file_access.h(218): note: see declaration of 'FileAccess::store_buffer'
scons: *** [modules\gdsdecomp\editor\gdre_editor.windows.editor.x86_64.obj] Error 2
scons: building terminated because of errors.
```

I'm able to build successfully on an earlier version of the source from just before v0.8.0: [https://github.com/godotengine/godot/commit/b31cc6960d20b758b2ad6d3a3d497a9d7964823e](https://github.com/godotengine/godot/commit/b31cc6960d20b758b2ad6d3a3d497a9d7964823e)

### Steps to reproduce

Building with `scons platform=windows` 
Using the latest Godot 4.4 source on master and the Visual Studio compiler




### Recovery log

n/a

### Link to game download

n/a

--- a/.github/workflows/all_builds.yml
+++ b/.github/workflows/all_builds.yml
@@ -26,7 +26,7 @@ on:
 env:
   GODOT_BASE_BRANCH: master
   # Change the README too
-  GODOT_MAIN_SYNC_REF: ec6a1c0e792ac8be44990749800a4654a293b9ee
+  GODOT_MAIN_SYNC_REF: 46c8f8c5c5874c7c56ea5b1384259de9402d3449
   SCONSFLAGS: verbose=yes warnings=all werror=no module_text_server_fb_enabled=yes minizip=yes deprecated=yes
   SCONSFLAGS_TEMPLATE: no_editor_splash=yes module_camera_enabled=no module_mobile_vr_enabled=no module_upnp_enabled=no module_websocket_enabled=no module_csg_enabled=yes module_gridmap_enabled=yes use_static_cpp=yes builtin_freetype=yes builtin_libpng=yes builtin_zlib=yes builtin_libwebp=yes builtin_libvorbis=yes builtin_libogg=yes disable_3d=no
   SCONS_CACHE_MSVC_CONFIG: true

--- a/README.md
+++ b/README.md
@@ -102,7 +102,7 @@ For ease of bootstrapping development, we have included launch, build, and setti
 
 ### Requirements
 
-Godot 4.0 (master branch) @ ec6a1c0e792ac8be44990749800a4654a293b9ee
+Godot 4.0 (master branch) @ 46c8f8c5c5874c7c56ea5b1384259de9402d3449
 
 - Support for building on 3.x has been dropped and no new features are being pushed
   - Godot RE Tools still retains the ability to decompile 3.x and 2.x projects, however.

--- a/compat/file_access_encrypted_v3.cpp
+++ b/compat/file_access_encrypted_v3.cpp
@@ -228,9 +228,9 @@ Error FileAccessEncryptedv3::get_error() const {
 	return eofed ? ERR_FILE_EOF : OK;
 }
 
-void FileAccessEncryptedv3::store_buffer(const uint8_t *p_src, uint64_t p_length) {
-	ERR_FAIL_COND_MSG(!writing, "File has not been opened in write mode.");
-	ERR_FAIL_COND(!p_src && p_length > 0);
+bool FileAccessEncryptedv3::store_buffer(const uint8_t *p_src, uint64_t p_length) {
+	ERR_FAIL_COND_V_MSG(!writing, false, "File has not been opened in write mode.");
+	ERR_FAIL_COND_V(!p_src && p_length > 0, false);
 
 	if (pos < get_length()) {
 		for (uint64_t i = 0; i < p_length; i++) {
@@ -243,6 +243,7 @@ void FileAccessEncryptedv3::store_buffer(const uint8_t *p_src, uint64_t p_length
 		}
 		pos += p_length;
 	}
+	return true;
 }
 
 void FileAccessEncryptedv3::flush() {
@@ -251,8 +252,8 @@ void FileAccessEncryptedv3::flush() {
 	// encrypted files keep data in memory till close()
 }
 
-void FileAccessEncryptedv3::store_8(uint8_t p_dest) {
-	ERR_FAIL_COND_MSG(!writing, "File has not been opened in write mode.");
+bool FileAccessEncryptedv3::store_8(uint8_t p_dest) {
+	ERR_FAIL_COND_V_MSG(!writing, false, "File has not been opened in write mode.");
 
 	if (pos < get_length()) {
 		data.write[pos] = p_dest;
@@ -261,6 +262,7 @@ void FileAccessEncryptedv3::store_8(uint8_t p_dest) {
 		data.push_back(p_dest);
 		pos++;
 	}
+	return true;
 }
 
 bool FileAccessEncryptedv3::file_exists(const String &p_name) {

--- a/compat/file_access_encrypted_v3.h
+++ b/compat/file_access_encrypted_v3.h
@@ -83,8 +83,8 @@ public:
 
 	virtual Error resize(int64_t p_length) override { return ERR_UNAVAILABLE; }
 	virtual void flush() override;
-	virtual void store_8(uint8_t p_dest) override; ///< store a byte
-	virtual void store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes
+	virtual bool store_8(uint8_t p_dest) override; ///< store a byte
+	virtual bool store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes
 
 	virtual bool file_exists(const String &p_name) override; ///< return true if a file exists
 

--- a/utility/file_access_apk.cpp
+++ b/utility/file_access_apk.cpp
@@ -391,11 +391,11 @@ void FileAccessAPK::flush() {
 	ERR_FAIL();
 }
 
-void FileAccessAPK::store_8(uint8_t p_dest) {
-	ERR_FAIL();
+bool FileAccessAPK::store_8(uint8_t p_dest) {
+	ERR_FAIL_V(false);
 }
-void FileAccessAPK::store_buffer(const uint8_t *p_src, uint64_t p_length) {
-	ERR_FAIL();
+bool FileAccessAPK::store_buffer(const uint8_t *p_src, uint64_t p_length) {
+	ERR_FAIL_V(false);
 } ///< store an array of bytes, needs to be overwritten by children.
 
 bool FileAccessAPK::file_exists(const String &p_name) {

--- a/utility/file_access_apk.h
+++ b/utility/file_access_apk.h
@@ -103,8 +103,8 @@ public:
 
 	virtual Error resize(int64_t p_length) override { return ERR_UNAVAILABLE; }
 	virtual void flush() override;
-	virtual void store_8(uint8_t p_dest) override; ///< store a byte
-	virtual void store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes, needs to be overwritten by children.
+	virtual bool store_8(uint8_t p_dest) override; ///< store a byte
+	virtual bool store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes, needs to be overwritten by children.
 
 	virtual bool file_exists(const String &p_name) override; ///< return true if a file exists
 

--- a/utility/file_access_gdre.cpp
+++ b/utility/file_access_gdre.cpp
@@ -301,13 +301,13 @@ void FileAccessGDRE::flush() {
 	return proxy->flush();
 }
 
-void FileAccessGDRE::store_8(uint8_t p_dest) {
-	ERR_FAIL_COND_MSG(proxy.is_null(), "File must be opened before use.");
+bool FileAccessGDRE::store_8(uint8_t p_dest) {
+	ERR_FAIL_COND_V_MSG(proxy.is_null(), false, "File must be opened before use.");
 	return proxy->store_8(p_dest);
 }
 
-void FileAccessGDRE::store_buffer(const uint8_t *p_src, uint64_t p_length) {
-	ERR_FAIL_COND_MSG(proxy.is_null(), "File must be opened before use.");
+bool FileAccessGDRE::store_buffer(const uint8_t *p_src, uint64_t p_length) {
+	ERR_FAIL_COND_V_MSG(proxy.is_null(), false, "File must be opened before use.");
 	return proxy->store_buffer(p_src, p_length);
 }
 

--- a/utility/file_access_gdre.h
+++ b/utility/file_access_gdre.h
@@ -114,8 +114,8 @@ public:
 	virtual String fix_path(const String &p_path) const override; ///< fix a path, i.e. make it absolute and in the OS format
 	virtual Error resize(int64_t p_length) override;
 	virtual void flush() override;
-	virtual void store_8(uint8_t p_dest) override; ///< store a byte
-	virtual void store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes
+	virtual bool store_8(uint8_t p_dest) override; ///< store a byte
+	virtual bool store_buffer(const uint8_t *p_src, uint64_t p_length) override; ///< store an array of bytes
 
 	virtual bool file_exists(const String &p_name) override; ///< return true if a file exists
 

